src/einsteinpy/bodies.py

```diff
@@ -15,195 +15,192 @@
+    ----------
+    name : string
+        Name or ID of the body
+    mass : ~astropy.units.kg
+        Mass of the body
+    q : ~astropy.units.C, optional
+        Charge on the body
+    R : ~astropy.units
+        Radius of the body
+    differential : ~einsteinpy.coordinates.differential.*, optional
+        Complete coordinates of the body
+    parent : Body, optional
+        The parent object of the body
+        Useful in case of multibody systems
     """
 
-    @u.quantity_input(mass=u.kg, R=u.km)
-    def __init__(
-        self,
-        name="Generic Body",
-        mass=0 * u.kg,
-        q=0 * u.C,
-        R=0 * u.km,
-        differential=None,
-        parent=None,
-    ):
-        """
-        Parameters
-        ----------
-        name : string
-            Name or ID of the body
-        mass : ~astropy.units.kg
-            Mass of the body
-        q : ~astropy.units.C, optional
-            Charge on the body
-        R : ~astropy.units
-            Radius of the body
-        differential : ~einsteinpy.coordinates.differential.*, optional
-            Complete coordinates of the body
-        parent : Body, optional
-            The parent object of the body
-            Useful in case of multibody systems
-        """
-        if differential:
-            if differential.system == "Cartesian":
-                self.pos_vec = [differential.x, differential.y, differential.z]
-                self.vel_vec = [differential.v_x, differential.v_y, differential.v_z]
+    name: str = "Generic Body"
+    mass: u.kg = 0 * u.kg
+    q: u.C = 0 * u.C
+    R: u.km = 0 * u.km
+    coords = None
+    parent: "Body" = None
+
+    def __post_init__(self):
+        diff = self.differential
+        if diff:
+            if diff.system == "Cartesian":
+                self.pos_vec = [
+                    diff.x,
+                    diff.y,
+                    diff.z,
+                ]
+                self.vel_vec = [
+                    diff.v_x,
+                    diff.v_y,
+                    diff.v_z,
+                ]
             else:
-                self.pos_vec = [differential.r, differential.theta, differential.phi]
-                self.vel_vec = [differential.v_r, differential.v_th, differential.v_p]
-        self.name = name
-        self.mass = mass
-        self.q = q
-        self.R = R
-        self.coords = differential
-        self.parent = parent
+                self.pos_vec = [
+                    diff.r,
+                    diff.theta,
+                    diff.phi,
+                ]
+                self.vel_vec = [
+                    diff.v_r,
+                    diff.v_th,
+                    diff.v_p,
+                ]
 
     def __str__(self):
-        return f"Body: ( Name: ({self.name}), Mass: ({self.mass}), Charge: ({self.q})', Radius: ({self.R}), \n \
-            Initial Coordinates: ({self.coords}), Parent Body: ({self.parent}) )"
+        return dedent(
+            f"""

```

Comment:
Indentation is improper. Make this exactly like here - https://github.com/einsteinpy/einsteinpy/pull/651#discussion_r1352033434.

Reply:
This was due `tox` messing with the indentation

Reply:
I'm getting an error with the Ci build after I changed it according to your suggestion @JeS24

Reply:
@VPraharsha03 It seems to be passing now, right?

Reply:
@JeS24 That's because I've used `tox` to reformat again

Reply:
Okay, cool.